// ngx-audio-player/esm2015/lib/component/base/base-audio-player-components.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { ViewChild, ElementRef } from "@angular/core";
export class BaseAudioPlayerFunctions {
    constructor() {
        this.loaderDisplay = false;
        this.isPlaying = false;
        this.currentTime = 0;
        this.duration = 0.01;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    currTimePosChanged(event) {
        this.player.nativeElement.currentTime = event.value;
    }
    /**
     * @return {?}
     */
    bindPlayerEvent() {
        this.player.nativeElement.addEventListener('playing', () => {
            this.isPlaying = true;
            this.duration = Math.floor(this.player.nativeElement.duration);
        });
        this.player.nativeElement.addEventListener('pause', () => {
            this.isPlaying = false;
        });
        this.player.nativeElement.addEventListener('timeupdate', () => {
            this.currentTime = Math.floor(this.player.nativeElement.currentTime);
        });
        this.player.nativeElement.addEventListener('loadstart', () => {
            this.loaderDisplay = true;
        });
        this.player.nativeElement.addEventListener('loadeddata', () => {
            this.loaderDisplay = false;
            this.duration = Math.floor(this.player.nativeElement.duration);
        });
    }
    ;
    /**
     * @return {?}
     */
    playBtnHandler() {
        if (this.loaderDisplay) {
            return;
        }
        if (this.player.nativeElement.paused) {
            this.player.nativeElement.play(this.currentTime);
        }
        else {
            this.currentTime = this.player.nativeElement.currentTime;
            this.player.nativeElement.pause();
        }
    }
    ;
    /**
     * @return {?}
     */
    play() {
        setTimeout(() => {
            this.player.nativeElement.play();
        }, 0);
    }
    ;
}
BaseAudioPlayerFunctions.propDecorators = {
    player: [{ type: ViewChild, args: ['audioPlayer',] }]
};
if (false) {
    /** @type {?} */
    BaseAudioPlayerFunctions.prototype.player;
    /** @type {?} */
    BaseAudioPlayerFunctions.prototype.timeLineDuration;
    /** @type {?} */
    BaseAudioPlayerFunctions.prototype.loaderDisplay;
    /** @type {?} */
    BaseAudioPlayerFunctions.prototype.isPlaying;
    /** @type {?} */
    BaseAudioPlayerFunctions.prototype.currentTime;
    /** @type {?} */
    BaseAudioPlayerFunctions.prototype.duration;
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
}
