// ngx-audio-player/esm5/lib/component/mat-basic-audio-player/mat-basic-audio-player.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { BaseAudioPlayerFunctions } from '../base/base-audio-player-components';
var MatBasicAudioPlayerComponent = /** @class */ (function (_super) {
    tslib_1.__extends(MatBasicAudioPlayerComponent, _super);
    function MatBasicAudioPlayerComponent() {
        var _this = _super.call(this) || this;
        _this.displayTitle = false;
        return _this;
    }
    /**
     * @return {?}
     */
    MatBasicAudioPlayerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.bindPlayerEvent();
    };
    /**
     * @return {?}
     */
    MatBasicAudioPlayerComponent.prototype.resetSong = /**
     * @return {?}
     */
    function () {
        this.player.nativeElement.src = this.audioUrl;
    };
    ;
    MatBasicAudioPlayerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'mat-basic-audio-player',
                    template: "<mat-card class=\"d-flex ngx-basic-audio-player\">\n    <audio [src]=\"audioUrl\" #audioPlayer></audio>\n    <button mat-button (click)='playBtnHandler();' class=\"ngx-basic-audio-player-button-handler\" style=\"font-size: 14px!important;\" [disabled]=\"loaderDisplay\">\n        <fa-icon *ngIf=\"loaderDisplay\" title=\"loading...\" class=\"fa fa-lg loading-track\" [pulse]=\"true\" [icon]=\"['fas','spinner']\"\n            aria-hidden=\"true\"></fa-icon>\n        <fa-icon *ngIf=\"!loaderDisplay && !isPlaying\" class=\"fa fa-lg play-track\" [icon]=\"['fas','play']\" aria-hidden=\"true\"></fa-icon>\n        <fa-icon *ngIf=\"!loaderDisplay && isPlaying\" class=\"fa fa-lg pause-track\" [icon]=\"['fas','pause']\" aria-hidden=\"true\"></fa-icon>\n    </button>\n    <div class=\"flex-fill pl-1 ngx-basic-audio-player-slider\">\n        <mat-slider style=\"width: 100%\" min=\"0\" max=\"{{duration}}\" value=\"{{currentTime}}\" (change)=\"currTimePosChanged($event)\"></mat-slider>\n    </div>\n    <div class=\"d-flex ngx-basic-audio-player-duration\">\n        <span *ngIf=\"duration !== 0.01\" class=\"pl-2 pr-3 my-auto\" style=\"font-size: 14px!important;\">\n            -{{duration-currentTime |\n            secondsToMinutes }}\n        </span>\n    </div>\n</mat-card>\n<mat-card *ngIf=\"displayTitle && title !== ''\" style=\"border-top: 1px solid #ddd;\">\n    <div style=\"text-align: center;\">\n        <div style=\"padding:1px 3px;\">\n            {{ title }}\n        </div>\n    </div>\n</mat-card>\n",
                    styles: ["@import url(https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.2.1/css/bootstrap-grid.min.css);mat-card,mat-slider{padding:0!important}button:focus,button:hover{outline:0!important}.ngx-basic-audio-player{min-width:275px}"]
                }] }
    ];
    /** @nocollapse */
    MatBasicAudioPlayerComponent.ctorParameters = function () { return []; };
    MatBasicAudioPlayerComponent.propDecorators = {
        title: [{ type: Input }],
        audioUrl: [{ type: Input }],
        displayTitle: [{ type: Input }]
    };
    return MatBasicAudioPlayerComponent;
}(BaseAudioPlayerFunctions));
export { MatBasicAudioPlayerComponent };
if (false) {
    /** @type {?} */
    MatBasicAudioPlayerComponent.prototype.title;
    /** @type {?} */
    MatBasicAudioPlayerComponent.prototype.audioUrl;
    /** @type {?} */
    MatBasicAudioPlayerComponent.prototype.displayTitle;
    /* Skipping unhandled member: ;*/
}
